ConsistencyManager/Protocols/ConsistencyManagerUpdatesListener.swift

```diff
@@ -0,0 +1,33 @@
+// © 2016 LinkedIn Corp. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import Foundation
+
+/**
+ This protocol allows a class to listen to ALL changes to the consistency manager.
+ */
+public protocol ConsistencyManagerUpdatesListener: class {
+    /**
+     Called whenever there is any change to the consistency manager.
+     This method is run on the main thread. If you have any extensive processing, it's highly recommended to do this on a background thread
+     since this will be called for every single consistency update.
+
+     - parameter consistencyManager: The consistency manager which has received the change.
+     - parameter model: The model which has been updated (NOTE: This model may have been deleted).
+     To check if it has been deleted, check `flattenedChildren[model.modelIdentifier] != nil`.
```

Comment:
This API is going to change in the next commit though to make it clearer. Optionals is a bit overloaded here. But changed.